python/cudf/cudf/core/column/datetime.py

```diff
@@ -274,14 +276,68 @@ def as_numerical_column(
     def as_string_column(
         self, dtype: Dtype, format=None, **kwargs
     ) -> "cudf.core.column.StringColumn":
+        need_names_formats = {
+            "%b",
+            "%B",
+            "%A",
+            "%a",
+        }
         if format is None:
             format = _dtype_to_format_conversion.get(
                 self.dtype.name, "%Y-%m-%d %H:%M:%S"
             )
+        if format in need_names_formats:
+
+            names = [
```

Comment:
We also don't need to redefine this list every time, can you make this also a global constant?